packages/core/src/tools/sanitize.ts

```diff
@@ -225,3 +227,25 @@ function tryToApplyToJSON(value: ExtendedContextValue) {
 
   return value
 }
+
+/**
+ * Helper function that displays a warning when the accumulated character count is over the limit
+ */
+function isOverCharacterLimit(
+  currentCharacterCount: number,
+  maxCharacterCount: number,
+  isDiscarded: boolean,
+  source: unknown
+) {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏we could really limit the responsibility of the helper to only displaying the warning

```ts
if (currentCharacterCount > maxCharacterCount) {
  warnOverCharacterLimit(maxCharacterCount, isDiscarded, source)
  ...
}
```

Reply:
👌